tests/assembleCoaddTestUtils.py

```diff
@@ -0,0 +1,625 @@
+        self.exposure = exposure
+        self.tract = tract
+        self.patch = patch
+        self.visit = visit
+
+    def get(self, bbox=None, datasetType=None, immediate=True, component=None, parameters=None):
+        """Retrieve the specified dataset using the API of the Gen 3 Butler.
+
+        Parameters
+        ----------
+        bbox : `lsst.geom.box.Box2I`, optional
+            If supplied, retrieve only a subregion of the exposure.
+        datasetType : `str`
+            Name of the type of exposure to retrieve.
+            Ignored, but supplied for compatibility with the Gen 3 Butler.
+        immediate : bool, optional
+            Ignored, but supplied for compatibility with the Gen 3 Butler.
+        component : `str`, optional
+            If supplied, return the named metadata of the exposure.
+        parameters : `dict`, optional
+            If supplied, use the parameters to modify the exposure,
+            typically by taking a subset.
+
+        Returns
+        -------
+        `lsst.afw.image.Exposure` or `lsst.afw.image.VisitInfo`
+        or `lsst.meas.algorithms.SingleGaussianPsf`
+            Either the exposure or its metadata, depending on the datasetType.
+        """
+        if component == 'psf':
+            return self.exposure.getPsf()
+        elif component == 'visitInfo':
+            return self.exposure.getInfo().getVisitInfo()
+        if parameters is not None:
+            if "bbox" in parameters:
+                bbox = parameters["bbox"]
+        exp = self.exposure.clone()
+        if bbox is not None:
+            return exp[bbox]
+        else:
+            return exp
+
+    @property
+    def dataId(self):
+        """Generate a valid data identifier.
+
+        Returns
+        -------
+        dataId : `lsst.daf.persistence.dataId`
+            Data identifier dict for the patch.
+        """
+        return DataId(tract=self.tract, patch=self.patch, visit=self.visit)
+
+    def datasetExists(self, tempExpName):
+        """Raise a more informative error if this Gen 2 method is called."""
+        raise NotImplementedError("Gen3 butler data references don't support `datasetExists`")
+
+
+def makeMockSkyInfo(bbox, wcs, patch):
+    """Construct a `Struct` containing the geometry of the patch to be coadded.
+
+    Parameters
+    ----------
+    bbox : `lsst.geom.Box`
+        Bounding box of the patch to be coadded.
+    wcs : `lsst.afw.geom.SkyWcs`
+        Coordinate system definition (wcs) for the exposure.
+
+    Returns
+    -------
+    skyInfo : `lsst.pipe.base.Struct`
+        Patch geometry information.
+    """
+    def getIndex():
+        return patch
+    patchInfo = pipeBase.Struct(getIndex=getIndex)
+    skyInfo = pipeBase.Struct(bbox=bbox, wcs=wcs, patchInfo=patchInfo)
+    return skyInfo
+
+
+class MockCoaddTestData:
+    """Generate repeatable simulated exposures with consistent metadata that
+    are realistic enough to test the image coaddition algorithms.
```

Comment:
I would rather keep it separate, since that code is all self contained and works differently. That code is set up for offline verification of the scientific validity of the algorithms, while this is meant for simple unit tests that are to be run regularly to catch bugs.